File: java/chainofresponsibility/AuthenticationMiddlewareHandler.java
package chainofresponsibility;

public class AuthenticationMiddlewareHandler extends AbstractMiddlewareHandler {

    @Override
    public boolean handle(Request request) {
        System.out.println("Checking if request is Authenticated...");
        if (!request.isAuthenticated()) {
            System.out.println("request is not authenticated");
            return false;
        }
        return super.handle(request);
    }
}

File: typescript/chainofresponsibility/authentication-middleware-handler.ts
import {AbstractMiddlewareHandler} from "./abstractMiddleware-handler";
import {ChainRequest} from "./request";

export class AuthenticationMiddlewareHandler extends AbstractMiddlewareHandler {
  public handle(request: ChainRequest): boolean {
    if (!request.isAuthenticated) {
      console.log(
        "AuthenticationMiddlewareHandler: User is not authenticated."
      );
      return false;
    }
    console.log("AuthenticationMiddlewareHandler: User is authenticated.");

    return super.handle(request);
  }
}
